### ISSUE TITLE
svite CLI option parsing does not honour config 

### ISSUE THREAD

**Issue by alexdilley** [dominikg/svite#13]
## Description

Default values defined by svite CLI options parsing (via Commander.js) overwrite equivalent values defined in `vite.config.js`.

## Reproduction

```js
// vite.config.js
module.exports = {
  assetsDir: '',
};
```

```sh
svite build
```

Sets `assetsDir` option to `_assets`.

## Additional Information

Removing the default values (and falling back to vite's defaults) with Commander.js has the consequence of setting any omitted option to `undefined`, so these will need to first be removed from the object merged into the config object resolved by vite's.

**Comment by dominikg** [comment]
thank you for reporting this issue.

**Comment by dominikg** [comment]
Argh, this is more complicated than i thought. 

I like to use default values in commanderjs because of the nice help output.
Ideally i could just resolve the config before starting the command and initialize the default values with the resolved config. 
This would be great for 2 reasons:
1) they automatically match vite defaults even if they change between vite versions
2) they also reflect user config values, meaning help would always show the value that is used if the cli option remains unset

Problem is, there are at least 3 options which require special care:
a) --config. Obviously you cannot resolve defaults for other options based on an option value ahead of time.
b) --mode. Can be 'development' or 'production' and has influence on vite-config/svite-plugin
c) --debug. Currently this is evaluated before 'vite' is even `require`d to be able to set the DEBUG env var before it it parsed by vite. If config was resolved before the commands, this would need to change.

So to keep defaults but not override config values unless the option deviates from default i think i'm going to have to setup defaults as objects and filter option values that match their given defaults. This would not have the benefit of 1) and 2) but avoid problems a-c.
